icaruscode/Timing/PMTTimingCorrectionsProvider.cxx

```diff
@@ -10,114 +10,92 @@
+   // Step 1) Add 1000000000 to the run number; e.g., run XXXXX -> 10000XXXXX
+   // Step 2) Multiply 1000000000
+   uint64_t runNum = uint64_t(run);
+   uint64_t timestamp = runNum+1000000000;
+   timestamp *= 1000000000;
 
+   if( fVerbose ) mf::LogInfo(fLogCategory) << "Run " << runNum << " corrections from DB timestamp " << timestamp << std::endl;
+   
+   return timestamp;
 }
 
-
-// -----------------------------------------------------------------------------
+// -------------------------------------------------------------------------------
 
 /// Function to look up the calibration database at the table holding the pmt hardware cables corrections
 void icarusDB::PMTTimingCorrectionsProvider::ReadPMTCablesCorrections( uint32_t run ) { 
 
     // pmt_cables_delay: delays of the cables relative to trigger 
     // and reset distribution
-    const std::string name("pmt_cables_delays_data");
-    Dataset dataset;
-    int error = ConnectToDataset( name, run, dataset );
-
-    if (error) throw(std::exception());
-
-    for( int row=4; row < getNtuples(dataset); row++ ) {
-
-        Tuple tuple = getTuple(dataset, row);
-
-        if( tuple != NULL ) {
-
-            unsigned int channel_id = getLongValue( tuple, 0, &error );
-            if( error ) throw std::runtime_error( "Encountered error while trying to access 'channel_id' on table " + name );
-            
-            // PPS reset correction
-            double reset_distribution_delay = getDoubleValue( tuple, 11, &error );
-            if( error ) throw std::runtime_error( "Encountered error '" + std::to_string(error) + "' while trying to access 'reset_distribution_delay' on table " + name );
-
-            // Trigger cable delay
-            double trigger_reference_delay = getDoubleValue( tuple, 10, &error );
-            if( error ) throw std::runtime_error( "Encountered error while trying to access 'trigger_reference_delay' on table " + name );
-
-            // Phase correction
-            double phase_correction = getDoubleValue( tuple, 13, &error );
-            if( error ) throw std::runtime_error( "Encountered error while trying to access 'phase_correction' on table " + name );
+    const std::string dbname("pmt_cables_delays_data");
+    lariov::DBFolder db(dbname, "", "", fTag, true, false);
+
+    bool ret = db.UpdateData( RunToDatabaseTimestamp(run) ); // select table based on run number   
+    if( !ret ) throw std::runtime_error( "Unable to find or open " + dbname + ".db");
+
+    std::vector<unsigned int> channelList;
+    ret = db.GetChannelList(channelList);
+
+    for( auto channel : channelList ) {
+        
+        // PPS reset correction
+        double reset_distribution_delay = 0;
+        int error  = db.GetNamedChannelData( channel, "reset_distribution_delay", reset_distribution_delay );
+        if( error ) throw std::runtime_error( "Encountered error while trying to access 'reset_distribution_delay' on table " + dbname );
```

Comment:
This is good.
Just mentioning that the current implementation always returns `0`.
Well, since it's still me and I can't be seen around saying that anything is good:
```suggestion
        if( error ) throw std::runtime_error( "Encountered error (code " + std::to_string(error) + ") while trying to access 'reset_distribution_delay' on table " + dbname );
```

Reply:
Yeah, these checks are currently useless... Anyway, adding your suggestion as well.